k2/csrc/fsa.h

```diff
@@ -77,13 +79,43 @@ struct Fsa {
   // contains the first arc-index leaving this state (index into `arcs`).
```

Comment:
It is NOT a good style to define a `struct` in c++ that has both public data members and public methods.

@qindazhu  @danpovey 
what do you think?

Reply:
Yes, no both public data members and methods, but one exception is const method and constructor. It's very convenient to make `arcs` and `arc_indexes` public (at least for now).

Actually people always to do this (public const methond and constructor, public members), such as define a Tree.